pkg/reconciler/knativeserving/knativeserving.go

```diff
@@ -92,18 +93,15 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, ks *servingv1alpha1.Knat
 
 	logger.Infow("Reconciling KnativeServing", "status", ks.Status)
 	stages := []func(context.Context, *mf.Manifest, *servingv1alpha1.KnativeServing) error{
+		r.transform,
```

Comment:
Line 79 has called r.transform once, why do you call it another time in stages?

Reply:
The other call is in the Finalize function, that's not part of the reconcile loop and doesn't go via this statement.